Framework/Nexus/inc/MantidNexus/NexusFileIO.h

```diff
@@ -205,10 +206,10 @@ bool NexusFileIO::writeNxValue(const std::string &name, const TYPE &value, const
   if (NXopendata(fileID, name.c_str()) == NX_ERROR)
     return false;
   for (unsigned int it = 0; it < attributes.size(); ++it) {
-    NXputattr(fileID, attributes[it].c_str(), (void *)avalues[it].c_str(), static_cast<int>(avalues[it].size() + 1),
-              NX_CHAR);
+    NXputattr(fileID, attributes[it].c_str(), reinterpret_cast<void *>(avalues[it].c_str()),
```

Comment:
Nice work on replacing the C-style casts. I think we probably want `static_cast` here though? I think we need to make sure the address is the same after the cast? - https://stackoverflow.com/a/573345

Reply:
I seem to get a build error when I use `static_cast`:
```
10:15:02 /jenkins_workdir/workspace/pull_requests-conda-linux/Framework/Nexus/inc/MantidNexus/NexusFileIO.h:274:47: error: invalid 'static_cast' from type 'const char*' to type 'void*'
10:15:02   274 |     NXputattr(fileID, attributes[it].c_str(), static_cast<void *>(avalues[it].c_str()),
```

Also it looks like elsewhere in this file it uses `reinterpret_cast` for similiar purposes

Reply:
Originally I went for `reinterpret_cast` because I asked chatGPT and it said:
```
reinterpret_cast:
If the cast is purely to change the type of the pointer without altering the underlying data, reinterpret_cast is appropriate. This is the direct replacement for a C-style cast to void*.

static_cast:
If value is already a pointer type, and you're converting it to void*, static_cast can be used. However, if value is not already a pointer type, this cast is not allowed for such conversions.
```